src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -684,19 +685,26 @@ internal sealed override bool IsInterface
             }
         }
 
-        internal override ManagedKind ManagedKind
+        internal override ManagedKind GetManagedKind(ref HashSet<DiagnosticInfo> useSiteDiagnostics)
         {
-            get
+            var managedKind = _flags.ManagedKind;
+            if (managedKind == ManagedKind.Unknown)
             {
-                var managedKind = _flags.ManagedKind;
-                if (managedKind == ManagedKind.Unknown)
-                {
-                    var baseKind = base.ManagedKind;
-                    _flags.SetManagedKind(baseKind);
-                    return baseKind;
-                }
-                return managedKind;
+                Debug.Assert(_managedKindUseSiteDiagnostics.IsDefault);
```

Comment:
This assertion does not pass reliably, probably because it is essentially an assertion that this thread is the first one to try and populate the ManagedKind on this symbol. This code works by basically letting threads that race to initialize this value overwrite one another, so the assertion is not actually valid.